examples/gpt-2/configs/config_model_117M.py

```diff
@@ -1,30 +1,30 @@
 """Texar config file of the GPT-2 model_117M model.
 """
+pretrained_model_name = "117M"
 
 vocab_size = 50257
-dim = 768
```

Comment:
I think it's better to define `dim=768` and use it (instead of `768`) in the following? This makes it clear all places using `dim` must be of the same value.

Reply:
Another question, is it even necessary to have these model config files at all? Our built-in `GPT2Decoder` should already have defined the hyperparameters, right (user only has to specify 117M or 345M)?

Reply:
The reason that I have the model config file is that we allow users to load their own checkpoint. They could write the model architecture in the model config file and set `pretrained_model_name` to be `None`. They then can use `torch.load()` to load their own checkpoint to their own defined model.

Reply:
The reason I deleted `dim=768` is that I use this model config file as `hparams` for `GPT2Decoder` directly. In the default `hparams` of `GPT2Decoder`, there is no corresponding field called `dim`.

Reply:
So it looks *loading (pretrained) model checkpoint* and *model arch config* are coupled? I.e., setting `pretrained_model_name` will not only define the model arch but also load the pretrained checkpoint. Can we decouple them, by having another hparam control whether to load pretrained checkpoint or not?